Release/CMakeLists.txt

```diff
@@ -15,14 +15,33 @@ set(CPPREST_VERSION_REVISION 18)
 
 enable_testing()
 
-set(WERROR ON CACHE BOOL "Treat Warnings as Errors.")
+if(MINGW)
+  # on mingw there are tons of warnings.
+  set(WERROR OFF CACHE BOOL "Treat Warnings as Errors.")
+else()
+  set(WERROR ON CACHE BOOL "Treat Warnings as Errors.")
+endif()
 set(CPPREST_EXCLUDE_WEBSOCKETS OFF CACHE BOOL "Exclude websockets functionality.")
 set(CPPREST_EXCLUDE_COMPRESSION OFF CACHE BOOL "Exclude compression functionality.")
 set(CPPREST_EXCLUDE_BROTLI ON CACHE BOOL "Exclude Brotli compression functionality.")
 set(CPPREST_EXPORT_DIR cmake/cpprestsdk CACHE STRING "Directory to install CMake config files.")
 set(CPPREST_INSTALL_HEADERS ON CACHE BOOL "Install header files.")
 set(CPPREST_INSTALL ON CACHE BOOL "Add install commands.")
 
+SET(DEFAULT_BUILD_TYPE "Release")
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to '${DEFAULT_BUILD_TYPE}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# There's a "string table overflow" with mingw-w64 when optimization is disabled.
+#
+# see also: https://stackoverflow.com/questions/14125007/gcc-string-table-overflow-error-during-compilation
+if(MINGW AND CMAKE_BUILD_TYPE STREQUAL "Debug" AND BUILD_SHARED_LIBS)
```

Comment:
link to a particular bug in gcc, so that we can revert the damage once the bug is fixed.